packages/sessions-sdk-react/src/components/sign-in-modal.tsx

```diff
@@ -363,31 +369,62 @@ const LimitsPage = ({
   const { whitelistedTokens, enableUnlimited, defaultRequestedLimits } =
     useSessionContext();
 
-  return (
-    <Page
-      heading="Session Limits"
-      message="Limit how many tokens this app is allowed to interact with"
-    >
-      <SessionLimits
-        enableUnlimited={enableUnlimited}
-        tokens={whitelistedTokens}
-        onSubmit={
-          sessionState.type === StateType.RequestingLimits
-            ? sessionState.submitLimits
-            : undefined
-        }
-        initialLimits={
-          (sessionState.type === StateType.RequestingLimits
-            ? sessionState.requestedLimits
-            : undefined) ??
-          defaultRequestedLimits ??
-          new Map()
-        }
-        // eslint-disable-next-line jsx-a11y/no-autofocus
-        autoFocus
-      />
-    </Page>
-  );
+  const state = useTokenAccountData(sessionState);
+
+  switch (state.type) {
+    case TokenDataStateType.Error:
+    case TokenDataStateType.Loaded: {
+      return (
+        <Page
+          heading="Session Limits"
+          message="Limit how many tokens this app is allowed to interact with"
+        >
+          <SessionLimits
+            enableUnlimited={enableUnlimited}
+            tokens={
+              state.type === TokenDataStateType.Error
+                ? []
+                : whitelistedTokens.filter((token) =>
+                    state.data.tokensInWallet.some((tokenInWallet) =>
+                      tokenInWallet.mint.equals(token),
+                    ),
+                  )
+            }
+            onSubmit={
+              sessionState.type === StateType.RequestingLimits
+                ? sessionState.submitLimits
+                : undefined
+            }
+            initialLimits={
+              (sessionState.type === StateType.RequestingLimits
+                ? sessionState.requestedLimits
+                : undefined) ??
+              defaultRequestedLimits ??
+              (state.type === TokenDataStateType.Loaded
+                ? new Map(
+                    state.data.tokensInWallet.map(
+                      ({
+                        mint,
+                        amountInWallet,
+                      }: {
+                        mint: PublicKey;
+                        amountInWallet: bigint;
+                      }) => [mint, amountInWallet],
+                    ),
```

Comment:
I don't think this will work correctly because it's going to show all the tokens a user has, but due to tx size limits we will fail if a user actually tries to set limits across them.  I'd stick with `new Map()` until we figure out a way to set more than a small number of token limits.